### PR TITLE
Bridge Developer Docs

### PR DESCRIPTION
**Pull request by VenelinMartinov**
Adds sphinx-based developer documentation to the bridge. Similar to https://github.com/pulumi/pulumi/pull/16953

Can be previewed here: https://pulumi-terraform-bridge.readthedocs.io/en/latest/docs/README.html

### CODE REVIEW

**Review by guineveresaenger — CHANGES_REQUESTED** [review]
Love it! 

I went through the preview and it looks like Sphinx wants us to put the markdown code fence tag into curlies: https://pulumi-developer-docs.readthedocs.io/latest/docs/diagrams.html#diagrams
which is honestly a bit frustrating because it means it won't render nicely on GitHub that way and now we have to choose where we want the nice diagram.

I lean towards putting the nice diagrams onto our nice docs site, though.

How is this published? Does it reflect latest? Do we deploy manually? if so, how?

finally - this may be beyond this tool, but is there any way this can be namespaced under the general Pulumi Developer Docs? Totally happy if the answer is "no, dev docs are repo scoped". Just curious!

**Comment by VenelinMartinov** [comment]
The developer docs site rebuilds on each commit, currently on this branch but I'll redirect it to master after merging. The credentials are in the password store.

This can indeed be a section under the main docs but we need some additional work there.

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Pulumi Terraform Bridge Developer Documentation
+
+Welcome to the Pulumi Terraform Bridge developer documentation! This site is designed to provide
+complete documentation for maintainers of and contributors to the Terraform Bridge as well as authors of Bridged Providers.
+
+:::{toctree}
+:maxdepth: 2
+:titlesonly:
+
```
Comment (iwahbe) [inline docs/README.md:9]: I think this needs to explain how to build the docs locally (so we can preview changes).

**Review by iwahbe — APPROVED** [review]
It would be amazing if we could build this into the same developer docs site that pu/pu uses.

File: .readthedocs.yaml
Comment (iwahbe) [inline .readthedocs.yaml:1]: I'm assuming that this is what lets the docs site build automatically.

**Comment by VenelinMartinov** [comment]
We've now added a link to the Bridge docs from the main docs and one from the Bridge docs to the main docs

**Review by guineveresaenger — APPROVED** [review]
Thank you for the adjustments!

**Pull request merged**